projects/plugins/jetpack/modules/simple-payments/simple-payments.php

```diff
@@ -149,15 +149,42 @@ function is_enabled_jetpack_simple_payments() {
 		return Jetpack::is_connection_ready() && Jetpack_Plan::supports( 'simple-payments' );
 	}
 
-	function parse_shortcode( $attrs, $content = false ) {
-		if ( empty( $attrs['id'] ) ) {
-			return;
+	/**
+	 * Get a WP_Post representation of a product
+	 *
+	 * @param int $id The ID of the product.
+	 *
+	 * @return array|false|WP_Post
+	 */
+	private function get_product( $id ) {
+		if ( ! $id ) {
+			return false;
 		}
-		$product = get_post( $attrs['id'] );
+
+		$product = get_post( $id );
 		if ( ! $product || is_wp_error( $product ) ) {
-			return;
+			return false;
 		}
 		if ( $product->post_type !== self::$post_type_product || 'publish' !== $product->post_status ) {
+			return false;
+		}
+		return $product;
+	}
+
+	/**
+	 * Creates the content from a shortcode
+	 *
+	 * @param array $attrs Shortcode attributes.
+	 * @param mixed $content unused.
+	 *
+	 * @return string|void
+	 */
+	public function parse_shortcode( $attrs, $content = false ) {
```

Comment:
You'll need to add a rule here so we can merge this:

```suggestion
	public function parse_shortcode( $attrs, $content = false ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```